### PR TITLE
feat: add missing fields for stage instance

### LINKED ISSUE #934: Add scheduled event field to creating stage instances
### Summary

Add a way to pass a relevant scheduled event when creating a stage instance.

### What is the feature request for?

The core library

### The Problem

Discord allows you to pass a scheduled event id when making a stage instance.

### The Ideal Solution

Allow passing a `ScheduledEvent` into `StageChannel.create_instance`
https://github.com/nextcord/nextcord/blob/22eb5d169f6a05f2c813d72a158972a0ba3d7dc1/nextcord/channel.py#L2022

### The Current Solution

_No response_

### Additional Context

discord/discord-api-docs#5687

### PR DESCRIPTION
**Pull request by Soheab**
## Summary

This PR adds some missing fields to stage instance.

Adds the following to `StageChannel.create_instance`:
- `send_start_notification`
- `scheduled_event`

And to `StageInstance`:
- `scheduled_event_id`
- `scheduled_event`

And to `HTTPClient.create_stage_instance`'s valid_keys:
- `send_start_notification`
- `guild_scheduled_event_id`

And to `StageInstance` (types/channel.py):
- `guild_scheduled_event_id`

Relevant API Docs:
https://discord.com/developers/docs/resources/stage-instance#create-stage-instance
`guild_scheduled_event_id` was added recently: https://github.com/discord/discord-api-docs/commit/67ce08b1e300c63e5a1e362838e21fb3fc7b1fcc to create

Fixes #934

## This is a **Code Change**

- [ ] I have tested my changes.
- [x] I have updated the documentation to reflect the changes.
- [x] I have run `task pyright` and fixed the relevant issues.

### CODE REVIEW

**Review by Skelmis — APPROVED** [review]

File: nextcord/stage_instance.py [hunk 1]
```diff
@@ -85,6 +95,17 @@ def channel(self) -> Optional[StageChannel]:
...
+    @cached_slot_property("_cs_scheduled_event")
+    def scheduled_event(self) -> Optional[ScheduledEvent]:
+        """Optional[:class:`ScheduledEvent`]: The scheduled event for this stage instance.
+
+        .. versionadded:: 2.7
+        """
+        if self.scheduled_event_id is None:
+            return None
+
+        return self._state.get_scheduled_event(self.scheduled_event_id)
```
Comment (xXenvy) [inline nextcord/stage_instance.py:107]:
```suggestion
        if self.scheduled_event_id is not None:
            return self._state.get_scheduled_event(self.scheduled_event_id)
```

**Review by xXenvy — COMMENTED** [review]
Also versionadded needs an update

File: nextcord/stage_instance.py (hunk 1 above)
Comment (teaishealthy) [reply nextcord/stage_instance.py:107]: I **think** a Ruff rule requires an explicit `return None`

**Review by EmmmaTech — CHANGES_REQUESTED** [review]
Looks good, just update all the `versionadded`s to `3.0`.

**Comment by Skelmis** [comment]
@EmmmaTech care to approve now the versionadded's are resolved?

**Review by EmmmaTech — APPROVED** [review]
looks good :3

**Pull request merged**